### PR TITLE
Return consolidated global styles in REST Request

### PR DESCRIPTION
**Pull request by oandregal**
Follow-up to https://github.com/WordPress/gutenberg/pull/29969

This PR covers a new use case for settings: REST Request. The API is going to be used by the mobile apps to get access to the data. Some data it needs is the consolidated global styles from core, theme, and user.

## How to test

- Load front, post-editor, and site editor with TT1-blocks & TwentyTwentyOne. Verify that global styles are respected (colors look as they should, etc).
- Go to `<your_site>/wp-json/__experimental/wp-block-editor/v1/settings` and verify that you have access to all the settings (including the ones at `__experimentalStyles`).

_Note: a quick way to test the endpoint is to add a return true; within get_items_permissions_check in the lib/class-wp-rest-block-editor-settings-controller.php file._

### CODE REVIEW

File: lib/global-styles.php [hunk 1]
```diff
@@ -76,60 +76,60 @@ function gutenberg_experimental_global_styles_enqueue_assets() {
...
 		WP_Theme_JSON_Resolver::theme_has_support() &&
 		gutenberg_supports_block_templates()
 	) {
+		// Only lookup for the user data if we need it.
+		$origin = 'user';
+	}
+	$consolidated = WP_Theme_JSON_Resolver::get_merged_data( $settings, $origin );
+
+	if ( 'rest-request' === $context ) {
+		$settings['__experimentalStyles'] = $consolidated->get_raw_data()['styles'];
```
Comment (geriux) [inline lib/global-styles.php:106]: Could this be:
```suggestion
		$settings['__experimentalGlobalStylesBaseStyles'] = $consolidated->get_raw_data();
```

To get the whole consolidated raw data and use the same name.

**Comment by geriux** [comment]
Hey @nosolosw thanks for making these changes!

Quick question, when I changed the default background color from `green` to `red`, in the response I get:

```json

{
    "styles": {
        "color": {
            "background": "var:preset|color|red",
            "text": "var(--wp--preset--color--dark-gray)"
        }
    }
}
```

Is the value of the background correct `var:preset|color|red`?

File: lib/global-styles.php (hunk 1 above)
Comment (oandregal) [reply lib/global-styles.php:106]: Was thinking of this:

- consolidated settings should be under `__experimentalFeatures` (which already are)
- consolidated styles should be under `__experimentalStyles` (this PR)
- are there any other fields meaningful for mobile (templateParts, etc)?

We also need to update `__experimentalGlobalStylesBaseStyles`: needs a better name (it passes down styles + settings) and filtering (some data shouldn't be there).  This is also not the consolidated data, but only the data for the theme, which I understand is not what mobile needs?

Comment (geriux) [reply lib/global-styles.php:106]: After double-checking, we are good with having `__experimentalStyles` and the consolidated settings under `__experimentalFeatures` 👍

**Comment by oandregal** [comment]
> Is the value of the background correct var:preset|color|red?

Yeah, that's expected. That's because the user selected a "preset" color from the theme palette. The value of that can be found under `settings.color.palette`, the color with the `red` slug. If the user had selected a raw value instead, something like `#0f5f42` would have been stored instead. Other values you can find there is whatever the theme author introduced in the `theme.json` ([see](https://github.com/WordPress/theme-experiments/blob/master/tt1-blocks/theme.json#L212) for an example).

**Comment by geriux** [comment]
> > Is the value of the background correct var:preset|color|red?
> 
> Yeah, that's expected. That's because the user selected a "preset" color from the theme palette. The value of that can be found under `settings.color.palette`, the color with the `red` slug. If the user had selected a raw value instead, something like `#0f5f42` would have been stored instead. Other values you can find there is whatever the theme author introduced in the `theme.json` ([see](https://github.com/WordPress/theme-experiments/blob/master/tt1-blocks/theme.json#L212) for an example).

Right, I guess I was expecting the value to be like:

`var(--wp--preset--color--red)`

Instead of

`var:preset|color|red`

Thought it might be a parsing issue there.

**Comment by oandregal** [comment]
:thinking: Now that I think of it: I believe we needed that format for the post editor because @jorgefilipecosta had found some issues using the CSS Custom Properties directly. I presume issues with KSES. However, we're switching away from variables in the post editor https://github.com/WordPress/gutenberg/issues/31488 so I wonder if we can remove that format: a lot less logic, indirection, and code to maintain. It looks like in the site editor we wouldn't need to use it anyway. Jorge, do you have thoughts on this?

**Review by geriux — APPROVED** [review]
This looks good to me! @nosolosw  I guess the pending [comment](https://github.com/WordPress/gutenberg/pull/31762#issuecomment-839932213) can be tackled in a separate PR?

**Comment by oandregal** [comment]
Tested this in various scenarios (post editor, site editor, rest) and works as expected.

File: lib/global-styles.php [hunk 2]
```diff
@@ -76,60 +76,60 @@ function gutenberg_experimental_global_styles_enqueue_assets() {
...
-	// from the theme support data.
-	$settings['__experimentalFeatures'] = $tree->get_settings();
-
-	// STEP 2 - IF EDIT-SITE, ADD DATA REQUIRED FOR GLOBAL STYLES SIDEBAR
-	//
-	// In the site editor, the user can change styles, so the client
-	// needs the ability to create them. Hence, we pass it some data
-	// for this: base styles (core+theme) and the ID of the user CPT.
+
+	if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
```
Comment (oandregal) [inline lib/global-styles.php:91]: I don't like this now. See https://github.com/WordPress/gutenberg/pull/29969/files#r631772812

**Pull request merged**